syntax = "proto3";

package messages;

// The request message containing the user's name.
// python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. ./messages.proto
message MessageToServer {
    string opcode = 1;
    string username = 2;
    string target = 3;
    string message = 4;
}

// The response messages containing the greetings
message MessageToClient {
    string usernameOrStatus = 1;
    string message = 2; 
}

// Done
message ServerLog {
    string message = 1; 
}


// The service definition.
service Server {
    // Sends a greeting
    rpc ReceiveMessageFromClient (MessageToServer) returns (ServerLog) {}
    rpc SendMessageToClient (MessageToServer) returns (MessageToClient) {}
    rpc LogOut (MessageToServer) returns (ServerLog) {}
    rpc LogIn (MessageToServer) returns (ServerLog) {}
    rpc QueueMessage (MessageToServer) returns (ServerLog) {}
    rpc CreateAccount (MessageToServer) returns (ServerLog) {}
    rpc DeleteAccount (MessageToServer) returns (MessageToClient) {}
    rpc SeeAllAccounts (MessageToServer) returns (MessageToClient) {}
}